### DISCUSSION TITLE
Null Reference error in WalletConnectUnity Windows build

### DISCUSSION THREAD

**Discussion by yozdev** [WalletConnect/walletconnect-monorepo#3912] [Developer Support]
Hello Team,

I am working on the latest commit https://github.com/WalletConnect/WalletConnectUnity/commit/ccb7c4bb8c15f1ce2e8efbb11059f2d8ff0dbbeb of WalletConnectUnity 2.0

While everything works smoothly in Editor, I get the following null reference error when I run a Windows build :

NullReferenceException: Object reference not set to an instance of an object.
WalletConnectSharp.Common.Utils.Extensions.ListenOnce (System.Object eventSource, System.String eventName, System.EventHandler handler) (at <00000000000000000000000000000000>:0)
WalletConnectSharp.Core.Controllers.Relayer+<>c__DisplayClass77_0.g__Task2|2 () (at <00000000000000000000000000000000>:0)
System.Runtime.CompilerServices.AsyncVoidMethodBuilder.Start[TStateMachine] (TStateMachine& stateMachine) (at <00000000000000000000000000000000>:0)
WalletConnectSharp.Core.Controllers.Relayer.TransportOpen (System.String relayUrl) (at <00000000000000000000000000000000>:0)
System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine] (TStateMachine& stateMachine) (at <00000000000000000000000000000000>:0)
WalletConnectSharp.Core.Controllers.Relayer.TransportOpen (System.String relayUrl) (at <00000000000000000000000000000000>:0)
WalletConnectSharp.Core.Controllers.Relayer.Init () (at <00000000000000000000000000000000>:0)
System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) (at <00000000000000000000000000000000>:0)
System.Runtime.CompilerServices.AsyncMethodBuilderCore+MoveNextRunner.Run () (at <00000000000000000000000000000000>:0)
System.Threading.Tasks.AwaitTaskContinuation.RunCallback (System.Threading.ContextCallback callback, System.Object state, System.Threading.Tasks.Task& currentTask) (at <00000000000000000000000000000000>:0)
System.Threading.Tasks.SynchronizationContextAwaitTaskContinuation.Run (System.Threading.Tasks.Task ignored, System.Boolean canInlineContinuationTask) (at <00000000000000000000000000000000>:0)
System.Threading.Tasks.Task.FinishContinuations () (at <00000000000000000000000000000000>:0)
System.Threading.Tasks.Task1[TResult].TrySetResult (TResult result) (at <00000000000000000000000000000000>:0) System.Runtime.CompilerServices.AsyncTaskMethodBuilder1[TResult].SetResult (TResult result) (at <00000000000000000000000000000000>:0)
WalletConnectSharp.Core.Controllers.Relayer.CreateProvider () (at <00000000000000000000000000000000>:0)
System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) (at <00000000000000000000000000000000>:0)
System.Runtime.CompilerServices.AsyncMethodBuilderCore+MoveNextRunner.Run () (at <00000000000000000000000000000000>:0)
System.Threading.Tasks.AwaitTaskContinuation.RunCallback (System.Threading.ContextCallback callback, System.Object state, System.Threading.Tasks.Task& currentTask) (at <00000000000000000000000000000000>:0)
System.Threading.Tasks.SynchronizationContextAwaitTaskContinuation.Run (System.Threading.Tasks.Task ignored, System.Boolean canInlineContinuationTask) (at <00000000000000000000000000000000>:0)
System.Threading.Tasks.Task.FinishContinuations () (at <00000000000000000000000000000000>:0)
System.Threading.Tasks.Task1[TResult].TrySetResult (TResult result) (at <00000000000000000000000000000000>:0) System.Runtime.CompilerServices.AsyncTaskMethodBuilder1[TResult].SetResult (TResult result) (at <00000000000000000000000000000000>:0)
WalletConnectSharp.Core.Controllers.Relayer.CreateProvider (System.String auth) (at <00000000000000000000000000000000>:0)
System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) (at <00000000000000000000000000000000>:0)
System.Runtime.CompilerServices.AsyncMethodBuilderCore+MoveNextRunner.Run () (at <00000000000000000000000000000000>:0)
System.Threading.Tasks.AwaitTaskContinuation.RunCallback (System.Threading.ContextCallback callback, System.Object state, System.Threading.Tasks.Task& currentTask) (at <00000000000000000000000000000000>:0)
System.Threading.Tasks.SynchronizationContextAwaitTaskContinuation.Run (System.Threading.Tasks.Task ignored, System.Boolean canInlineContinuationTask) (at <00000000000000000000000000000000>:0)
System.Threading.Tasks.Task.FinishContinuations () (at <00000000000000000000000000000000>:0)
System.Threading.Tasks.Task1[TResult].TrySetResult (TResult result) (at <00000000000000000000000000000000>:0) System.Threading.Tasks.TaskCompletionSource1[TResult].TrySetResult (TResult result) (at <00000000000000000000000000000000>:0)
WalletConnectUnity.Utils.MTQ.Update () (at <00000000000000000000000000000000>:0)
--- End of stack trace from previous location where exception was thrown ---
System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.b__7_0 (System.Object state) (at <00000000000000000000000000000000>:0)
UnityEngine.UnitySynchronizationContext+WorkRequest.Invoke () (at <00000000000000000000000000000000>:0)
UnityEngine.UnitySynchronizationContext.Exec () (at <00000000000000000000000000000000>:0)

I get the above mentioned error after the following debug log message :

[Relayer] Opening transport
UnityEngine.Debug:Log(Object)
WalletConnectUnity.Utils.MTQ:Update()

I even tested building the sample Unity demo provided, but still I get the same error.

1. Make a Windows build of the SimpleUnityDapp scene
2. Run the demo
3. See error

Screenshots

![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/113970224/9f1d68d1-e679-43c7-81cd-9686ebfc385e)

Environment

OS: Windows 10
Unity version 2022.3.10f1
WalletConnectUnity version/commit https://github.com/WalletConnect/WalletConnectUnity/commit/ccb7c4bb8c15f1ce2e8efbb11059f2d8ff0dbbeb

Please note this is happening on Windows build.

Wallets used for testing
Metamask

Please provide solution.

Thanks

**Comment by glitch-txs** [comment]
Hi, please open an issue in the Unity repository
